icu4c/source/i18n/number_longnames.cpp

```diff
@@ -295,17 +502,24 @@ void LongNameHandler::forCompoundUnit(const Locale &loc, const MeasureUnit &unit
         status = U_INTERNAL_PROGRAM_ERROR;
         return;
     }
+
+    DerivedComponents derivedPerCases(loc, "case", "per", unitDisplayCase);
+
     UnicodeString primaryData[ARRAY_LENGTH];
-    getMeasureData(loc, unit, width, primaryData, status);
+    getMeasureData(loc, unit, width, derivedPerCases.value0(), primaryData, status);
     if (U_FAILURE(status)) {
         return;
     }
     UnicodeString secondaryData[ARRAY_LENGTH];
-    getMeasureData(loc, perUnit, width, secondaryData, status);
+    getMeasureData(loc, perUnit, width, derivedPerCases.value1(), secondaryData, status);
     if (U_FAILURE(status)) {
         return;
     }
 
+    // TODO(icu-units#139): implement these rules:
+    // <deriveComponent feature="plural" structure="per" ...>
+    // This has impact on multiSimpleFormatsToModifiers(...) below too.
+    // These rules are currently (ICU 69) all the same and hard-coded below.
     UnicodeString perUnitFormat;
```

Comment:
Nit:
```
        // TODO(icu-units#139): implement these rules:
        //    - <deriveComponent feature="plural" structure="per" ...>
        //    - This has impact on multiSimpleFormatsToModifiers(...) below too.
        // 
        // These rules are currently (ICU 69) all the same and hard-coded below.
```

Reply:
This is still a TODO, even after my next PR, because it does not impact ICU 69 with CLDR 39.